sharding/sharding.go

```diff
@@ -78,7 +78,12 @@ func NewFerry(config *Config) (*ShardingFerry, error) {
 }
 
 func (r *ShardingFerry) Initialize() error {
-	return r.Ferry.Initialize()
+	err := r.Ferry.Initialize()
+	if err != nil {
+		r.Ferry.ErrorHandler.Fatal("ferry.initialize", err)
```

Comment:
Is it necessary to call the error handler and panic here? Until now, I believe we only use the error handler during the Run() phase of ghostferry, where it is not possible to directly return an error.

If the idea is to give an error callback, we expose that API and/or hook up the logrus error reporting as a part of #47.

Reply:
The idea is exactly giving our application the callback. We still don't have #47 solved, so we'll have to live with this for a bit